homeassistant/components/roborock/__init__.py

```diff
@@ -134,6 +135,28 @@ async def on_stop(_: Any) -> None:
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
+    device_registry = dr.async_get(hass)
+    device_entries = dr.async_entries_for_config_entry(
+        device_registry, config_entry_id=entry.entry_id
+    )
+    for device in device_entries:
+        for identifier in device.identifiers:
```

Comment:
This requires that all identifiers for a device are present in the device map. While a device in practice only has one identifier, I think its more correct to check if *any* identifier for the device is present.


How about something like this:
```
    for device in device_entries:
        device_duids = set(identifier[1].replace("_dock", "") for identifier in device.identifiers)
        if any(device_duid in device_map for device_duid in device_duids):
            continue
        # <.... then remove device>
```